Utilities/bootstrap

```diff
@@ -167,14 +167,24 @@ def add_test_args(parser):
 
 def parse_global_args(args):
     """Parses and cleans arguments necessary for all actions."""
-    args.build_dir = os.path.abspath(args.build_dir)
-    args.project_root = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
-    args.tsc_source_dir = os.path.join(args.project_root, "..", "swift-tools-support-core")
-    args.yams_source_dir = os.path.join(args.project_root, "..", "yams")
-    args.swift_argument_parser_source_dir = os.path.join(args.project_root, "..", "swift-argument-parser")
-    args.swift_driver_source_dir = os.path.join(args.project_root, "..", "swift-driver")
-    args.swift_crypto_source_dir = os.path.join(args.project_root, "..", "swift-crypto")
-    args.source_root = os.path.join(args.project_root, "Sources")
+    # Test if 'build_dirs' and 'source_dirs' exist, and initialise them only if not.
+    # Otherwise, both are reset to empty dictionaries every time 'parse_global_args' is called, which crashes 'test', because 'test' calls it (via 'parse_test_args' → 'parse_build_args') after 'build' has called it (via 'parse_build_args').
```

Comment:
Looking like the non-ASCII character here is causing the CI error:
```
  File "/home/buildnode/jenkins/workspace/swift-package-manager-Linux-smoke-test/branch-main/swiftpm/Utilities/bootstrap", line 171
SyntaxError: Non-ASCII character '\xe2' in file /home/buildnode/jenkins/workspace/swift-package-manager-Linux-smoke-test/branch-main/swiftpm/Utilities/bootstrap on line 171, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details
```

Reply:
The non-ASCII character is `→` (U+2192). I should have tested after writing that line of comment, instead of assuming it would work. I suppose it's because I'm too used to unicode working out-of-the-box in Swift.

There are 2 solutions in my opinion:

1. Replace "→" with "->".
2. Add `# encoding: utf-8` to the second line of the file (according to [PEP 263](https://www.python.org/dev/peps/pep-0263/)).

Both of them work in my local testing, but I prefer #2, because it's more future-proof. What do you think?

Reply:
Alternatively, I can move `build_dirs` and `source_dirs`'s initialisation to `main`, and remove the comment.

Reply:
Adding the encoding specifier sounds preferable to me, just to avoid folks making the same assumption as you which is pretty natural coming from Swift.

Reply:
I agree, adding the encoding seems reasonable.  Thanks!

As for the initialization, I don't have a strong preference either way.  I like the robustness of initializing them on first use, but kind of dislike the wordiness.  So whatever you prefer is fine here, I think.

Reply:
I added the encoding specifier line in 99608ff01453c25d0630d19568a459dd81db19da.

I left `build_dirs` and `source_dirs` as conditionally initialised, because it _feels like_ that it's easier to move them outwards if needed in the future than inwards.

Reply:
That sounds like a good choice.  Thanks for your work on this!